Sources/MongoSwift/BSON/BSONValue.swift

```diff
@@ -61,12 +61,20 @@ public protocol BSONValue {
     * - Parameters:
     *   - storage: A `DocumentStorage` to write to.
     *   - key: A `String`, the key under which to store the value.
+    *
+    * - Throws:
+    *   - `RuntimeError.internalError` if the document would become too large by encoding this key-value pair.
+    *   - `UserError.invalidArgumentError` in some cases if this `BSONValue` was modified to a state that cannot be
```

Comment:
is this ever anything besides the too large case? 
if not, that seems like more a problem with the `DocumentStorage` than the `BSONValue` itself, so maybe we can rephrase.
I wonder if it is worth providing a way for users to check document size.... do any other drivers do that?

Reply:
makes sense. let's not bother exposing unless someone asks for it then 🙂

Reply:
oops, looking at the diff now, I was actually referring to the second error here - "in some cases if this `BSONValue` was modified to a state that cannot be encoded to BSON"

Reply:
perfect 👍